include/pinocchio/spatial/inertia.hpp

```diff
@@ -424,6 +428,37 @@ namespace pinocchio
       return *this;
     }
 
+    InertiaTpl __minus__(const InertiaTpl & Yb) const
+    {
+      /* Y_{a-b} = ( m_a-m_b,
+       *             (m_a*c_a - m_b*c_b ) / (m_a - m_b),
+       *             I_a - I_b + (m_a*m_b)/(m_a-m_b) * AB_x * AB_x )
+       */
+
+      const Scalar eps = ::Eigen::NumTraits<Scalar>::epsilon();
+
+      const Scalar & mab = mass()-Yb.mass();
```

Comment:
It would be nice to check whether the resulting inertia remains physically consistent, certainly with an assert.

Reply:
I have included asserts for checking mass is positive (here and in FromDynamicParameters). However, I wonder what is the expected design for this class. Should `InertialTpl` handle fully physically consistent only? Or could it accept "partial" physical consistency or anything?

For instance, the `Random` function generates positive masses and rotational inertias with positive principal inertia components. However, this is not what we refer to as *fully physical consistency*. This means also having positive second moments of inertia, i.e., satisfying the named "triangular inequalities". In short, **Random function doesn't support full physical consistency, just a partial physical consistency**.

Moreover, `FromDynamicParameters` doesn't assert positive masses or rotation inertia triangular inequalities. In short, **FromDynamicParameters function supports any type of parameter including the ones that don't satisfy physics**.

In conclusion, I am inclined 
  1. to allow any time of parameters, 
  2. to remove asserts (as requested and introduced in this PR), and
  3. to define a "checkConsistency" function where users can trigger their checks on demand.

Please let me know what are your thoughts and how to proceed with this PR.

Reply:
@jcarpent -- please let me know your thoughts. I still think we should not include these asserts as resulting operators might not necessarily satisfy fully physical consistency. The reason is this class should change much more to do that. We can arrange a meeting to discuss this.

Be aware that the unit tests are not passing because they are designed to test conditions that don't meet full physical consistency. I'll need to modify them if we want to do that.